--- a/src/calibre/utils/podofo/outline.cpp	2026-09-18 22:34:39.014559342 +0200
+++ b/src/calibre/utils/podofo/outline.cpp	2026-09-18 22:34:39.619991218 +0200
@@ -61,12 +61,12 @@
             PyErr_Format(PyExc_ValueError, "Invalid page number: %u", num);
             return NULL;
         }
-        auto dest = self->doc->CreateDestination();
-        dest->SetDestination(*page, left, top, zoom);
+        auto dest = std::make_shared<PdfDestination>(*page, left, top, zoom);
         if (PyObject_IsTrue(as_child)) {
-            ans->item = &self->item->CreateChild(title);
-        } else ans->item = &self->item->CreateNext(title);
-        ans->item->SetDestination(*dest);
+            ans->item = self->item->CreateChild(title, dest);
+        } else {
+            ans->item = self->item->CreateNext(title, dest);
+        }
     } catch (const PdfError &err) {
         podofo_set_exception(err);
         return NULL;
