diff -up calibre-8.16.2/setup/translations.py.6~ calibre-8.16.2/setup/translations.py
--- calibre-8.16.2/setup/translations.py.6~	2025-12-06 04:52:49.000000000 +0100
+++ calibre-8.16.2/setup/translations.py	2026-01-12 18:33:54.616172671 +0100
@@ -10,6 +10,7 @@ import glob
 import hashlib
 import inspect
 import json
+import optparse
 import os
 import re
 import shlex
@@ -22,6 +23,7 @@ import time
 from collections import defaultdict
 from functools import lru_cache, partial
 from locale import normalize as normalize_locale
+from contextlib import suppress
 
 from setup import Command, __appname__, __version__, build_cache_dir, dump_json, edit_file, is_ci, require_git_master
 from setup.iso_codes import iso_data
@@ -284,6 +286,10 @@ class Translations(POT):  # {{{
     DEST = os.path.join(os.path.dirname(POT.SRC), 'resources', 'localization',
             'locales')
 
+    def add_options(self, parser):
+        with suppress(optparse.OptionConflictError):  # ignore if option already added
+            parser.add_option('--path-to-isocodes', help='Path to previously downloaded iso-codes-main.zip')
+
     @property
     def cache_dir(self):
         ans = self.j(build_cache_dir(), 'translations')
@@ -333,6 +339,7 @@ class Translations(POT):  # {{{
         return locale, os.path.join(self.DEST, locale, 'messages.mo')
 
     def run(self, opts):
+        iso_data.run(opts)
         self.compile_main_translations()
         self.compile_content_server_translations()
         self.freeze_locales()
@@ -798,12 +805,17 @@ class ISO639(Command):  # {{{
     DEST = os.path.join(os.path.dirname(POT.SRC), 'resources', 'localization',
             'iso639.calibre_msgpack')
 
+    def add_options(self, parser):
+        with suppress(optparse.OptionConflictError):  # ignore if option already added
+            parser.add_option('--path-to-isocodes', help='Path to previously downloaded iso-codes-main.zip')
+
     def run(self, opts):
         dest = self.DEST
         base = self.d(dest)
         if not os.path.exists(base):
             os.makedirs(base)
         self.info('Packing ISO-639 codes to', dest)
+        iso_data.run(opts)
         root = json.loads(iso_data.db_data('iso_639-3.json'))
         entries = root['639-3']
         by_2 = {}
@@ -857,12 +869,17 @@ class ISO3166(ISO639):  # {{{
     DEST = os.path.join(os.path.dirname(POT.SRC), 'resources', 'localization',
             'iso3166.calibre_msgpack')
 
+    def add_options(self, parser):
+        with suppress(optparse.OptionConflictError):  # ignore if option already added
+            parser.add_option('--path-to-isocodes', help='Path to previously downloaded iso-codes-main.zip')
+
     def run(self, opts):
         dest = self.DEST
         base = self.d(dest)
         if not os.path.exists(base):
             os.makedirs(base)
         self.info('Packing ISO-3166 codes to', dest)
+        iso_data.run(opts)
         db = json.loads(iso_data.db_data('iso_3166-1.json'))
         codes = set()
         three_map = {}
