--- calibre/src/calibre/gui2/update.py~	2011-09-09 10:46:16.000000000 -0600
+++ calibre/src/calibre/gui2/update.py	2011-09-13 13:39:52.629503202 -0600
@@ -120,13 +120,13 @@
 
 class UpdateMixin(object):
 
-    def __init__(self, opts):
-        self.last_newest_calibre_version = NO_CALIBRE_UPDATE
-        if not opts.no_update_check:
-            self.update_checker = CheckForUpdates(self)
-            self.update_checker.update_found.connect(self.update_found,
-                    type=Qt.QueuedConnection)
-            self.update_checker.start()
+    #def __init__(self, opts):
+        #self.last_newest_calibre_version = NO_CALIBRE_UPDATE
+        #if not opts.no_update_check:
+        #    self.update_checker = CheckForUpdates(self)
+        #    self.update_checker.update_found.connect(self.update_found,
+        #            type=Qt.QueuedConnection)
+        #    self.update_checker.start()
 
     def recalc_update_label(self, number_of_plugin_updates):
         self.update_found('%s%s%d'%(self.last_newest_calibre_version, VSEP,
