--- a/setup/extensions.py	2011-12-29 15:00:50.833733423 +0100
+++ b/setup/extensions.py	2011-12-29 15:01:15.196670817 +0100
@@ -429,6 +429,7 @@ class Build(Command):
                     makefile=mf, universal=OSX_SDK, qt=1)
             makefile.extra_lflags = qt_objects
             makefile.extra_include_dirs = ext.inc_dirs
+            makefile.LIBS.append('-lpython'+sysconfig.get_python_version())
             makefile.generate()
 
             self.check_call([make, '-f', mf], cwd=src_dir)
