diff -up calibre-7.13.0/setup/iso_codes.py.6~ calibre-7.13.0/setup/iso_codes.py
--- calibre-7.13.0/setup/iso_codes.py.6~	2024-06-28 05:34:16.000000000 +0200
+++ calibre-7.13.0/setup/iso_codes.py	2024-06-28 14:56:53.520829142 +0200
@@ -15,7 +15,7 @@ from setup import Command, download_secu
 
 class ISOData(Command):
     description = 'Get ISO codes name localization data'
-    URL = 'https://salsa.debian.org/iso-codes-team/iso-codes/-/archive/main/iso-codes-main.zip'
+    URL = '/tmp/iso-codes-main.zip'
 
     def add_options(self, parser):
         with suppress(optparse.OptionConflictError):  # ignore if option already added
@@ -31,7 +31,7 @@ class ISOData(Command):
                 assert len(top) == 1
                 self.top_level = top.pop()
             else:
-                self._zip_data = BytesIO(download_securely(self.URL))
+                self._zip_data = open(self.URL, "rb")
 
     def __init__(self):
         super().__init__()
@@ -51,7 +51,7 @@ class ISOData(Command):
         name = name.split('.', 1)[0]
         pat = f'{self.top_level}/{name}/*.po'
         if self.zip_data is None:
-            self._zip_data = BytesIO(download_securely(self.URL))
+            self._zip_data = open(self.URL, "rb")
         with zipfile.ZipFile(self.zip_data) as zf:
             for name in fnmatch.filter(zf.namelist(), pat):
                 dest = os.path.join(output_dir, name.split('/')[-1])
