Index: calibre-4.7.0/setup/build_environment.py
===================================================================
--- calibre-4.7.0/setup/build_environment.py
+++ calibre-4.7.0/setup/build_environment.py	2019-12-27 09:56:41.955908244 +0100
@@ -87,7 +87,7 @@
 qt = {x:readvar(y) for x, y in {'libs':'QT_INSTALL_LIBS', 'plugins':'QT_INSTALL_PLUGINS'}.items()}
 qmakespec = readvar('QMAKE_SPEC') if iswindows else None
 
-pyqt['sip_bin'] = os.environ.get('SIP_BIN', 'sip')
+pyqt['sip_bin'] = os.environ.get('SIP_BIN', 'python3-sip')
 
 import PyQt5
 from PyQt5.QtCore import PYQT_CONFIGURATION
@@ -108,7 +108,7 @@
         else:
             q = os.path.join(os.path.dirname(PyQt5.__file__), 'bindings')
             if not os.path.exists(q):
-                q = os.path.join(sys.prefix, 'share', 'sip')
+                q = os.path.join(sys.prefix, 'share', 'python3-sip')
     q = os.environ.get('SIP_DIR', q)
     for x in ('', 'Py2-PyQt5', 'PyQt5', 'sip/PyQt5'):
         base = os.path.join(q, x)
